src/primitives/transaction.cpp

```diff
@@ -162,3 +162,12 @@ std::string CTransaction::ToString() const
         str += "    " + tx_out.ToString() + "\n";
     return str;
 }
+
+bool CTransaction::IsConfidential() const
+{
+    for (const CTxOut& out : vout) {
+        if (out.IsFee()) continue;
+        if (out.nValue.IsExplicit() || out.nAsset.IsExplicit()) return false;
```

Comment:
is this the right behavior: a transaction can have both confidential and unconfidential outputs, does this mean that to obtain the discount would suffice to have a single output being blinded then?

Reply:
the way it currently works here is that all outputs in the transaction must be blinded (except fee output)